src/views/inline_footnote.ts

```diff
@@ -217,18 +217,22 @@ export class InlineFootnoteView
   }
 
   handleInsert = (footnotes: FootnoteNode[]) => {
+    const pos = this.getPos()
+    const tr = this.view.state.tr
+
     if (footnotes.length) {
-      const pos = this.getPos()
       const rids = footnotes.map((note) => note.attrs.id)
-      const tr = this.view.state.tr
       if (rids.length) {
         tr.setNodeAttribute(pos, 'rids', rids)
-        this.view.dispatch(tr)
       } else if (isPendingInsert(this.node)) {
         tr.delete(pos, pos + this.node.nodeSize)
-        this.view.dispatch(tr)
       }
+    } else {
+      // When no footnotes, set rids to an empty array
```

Comment:
When the rids are empty, we should _delete_ the inline footnote.

Reply:
Got it, but I have a few questions. What exactly happens when deleting an inline footnote (when `rids === 0`)? Does it delete just the inline footnote, remove the citation, or delete the reference footnote as well? From the current implementation, it seems that deleting inline footnotes occurs as part of deleting the footnote itself, but the reverse case (deleting the inline footnote separately) isn’t covered.

In the scenario I encountered ( **please check the attached record**) , deleting the inline footnote with `rids === 0` and `selectedFootnotes === 0` results in the inline footnote being deleted and the reference footnote becoming uncited after approving the suggestion. The `uncite` logic is handled in the `updateContent` method of the footnote view, which triggers on `setNodeMarkup` ( as previously implemented on this PR) and similar actions, but not on deletion. 

If we plan to handle this by unciting the footnote, we may need an alternative approach for managing the unciting logic, possibly outside `footnote.view` or by using a different method. Let me know if you have any suggestions, ideas, or thoughts on this.
[inlinefootnotes.webm](https://github.com/user-attachments/assets/87d70d70-6f47-4857-8451-cea0a947002c)

Reply:
We should just delete the inline footnote, without touching the actual footnote. This is the same as deleting the inline footnote using backspace.

Reply:
done